src/script/InitDelegateeDeposits.s.sol

```diff
@@ -0,0 +1,152 @@
+
+    bytes[] memory _data = new bytes[](_delegateeAddressesLength);
+    for (uint256 i = 0; i < _delegateeAddressesLength; i++) {
+      _data[i] = abi.encodeWithSelector(IGovLst.depositForDelegatee.selector, _delegateeAddresses[i]);
+    }
+
+    bytes[] memory _results = govLst.multicall(_data);
+    for (uint256 i = 0; i < _delegateeAddressesLength; i++) {
+      _depositIds[i] = abi.decode(_results[i], (uint256));
+    }
+    return _depositIds;
+  }
+
+  /// @notice Filters delegatee addresses to identify those needing deposit initialization.
+  /// @param _delegateeAddresses Array of all delegatee addresses.
+  /// @param _depositIds Array of deposit IDs corresponding to each delegatee address.
+  /// @return _addressesToInit An array of addresses that need initialization.
+  /// @return _numOfAddressesToInit The count of addresses that need initialization.
+  function filterDelegateeAddresses(address[] memory _delegateeAddresses, uint256[] memory _depositIds)
+    public
+    view
+    virtual
+    returns (address[] memory, uint256)
+  {
+    address[] memory _addressesToInit = new address[](_delegateeAddresses.length);
+    uint256 _numOfAddressesToInit = 0;
+
+    for (uint256 i = 0; i < _delegateeAddresses.length; i++) {
+      if (_depositIds[i] == 0 || _depositIds[i] == DEFAULT_DEPOSIT_ID) {
+        _addressesToInit[_numOfAddressesToInit] = _delegateeAddresses[i];
+        _numOfAddressesToInit++;
+      }
+    }
+    return (_addressesToInit, _numOfAddressesToInit);
+  }
+
+  /// @notice Initializes deposits for delegatee addresses in batches using multicall.
+  /// @param _addressesToInit Array of delegatee addresses that need initialization.
+  /// @param _numOfAddressesToInit Number of addresses to initialize.
+  /// @return _allResults An array of results from the initialization calls.
+  /// @return _batchCount The number of batches processed.
+  function callFetchOrInitializeDepositForDelegatee(address[] memory _addressesToInit, uint256 _numOfAddressesToInit)
+    public
+    virtual
+    returns (bytes[] memory, uint256)
+  {
+    uint256 _totalAddresses = _numOfAddressesToInit;
+    uint256 _batchCount = (_totalAddresses + BATCH_SIZE - 1) / BATCH_SIZE;
+
+    bytes[] memory _allResults = new bytes[](_totalAddresses);
+
+    for (uint256 _batchIndex = 0; _batchIndex < _batchCount; _batchIndex++) {
+      uint256 _startIndex = _batchIndex * BATCH_SIZE;
+      uint256 _endIndex = _startIndex + BATCH_SIZE;
+      if (_endIndex > _totalAddresses) {
+        _endIndex = _totalAddresses;
+      }
+
+      uint256 _currentBatchSize = _endIndex - _startIndex;
+      bytes[] memory _batchData = new bytes[](_currentBatchSize);
+
+      for (uint256 i = 0; i < _currentBatchSize; i++) {
+        _batchData[i] = abi.encodeWithSelector(
+          IGovLst.fetchOrInitializeDepositForDelegatee.selector, _addressesToInit[_startIndex + i]
+        );
+      }
+
+      bytes[] memory _batchResults = govLst.multicall(_batchData);
```

Comment:
This line needs to be broadcast, right?

Reply:
I don't think so because we have `startBroadcast` in `run`, which calls this function.

Reply:
Ahh ok, missed that